### ISSUE TITLE
Fixed IP(s) for loadbalancer

### ISSUE THREAD

**Issue by kiorky** [ovh/public-cloud-roadmap#209]
As a managed kubernetes user, I want to be able to link one or more IPs to my loadbalancer service.

The bonus would be that i can tie any IPFO belonging to my OVH account to that specific loadbalancer.

Goal is to route incoming trafic to a stable IP endpoint that can be remapped to another endpoint whenever i want and for whatever reason i have. 

This is tied to #103 and i think we need both.

**Comment by mhurtrel** [comment]
Hello @kiorky 
I agree this is different from #103 
I just want to highlight that load balancer have a fixed IP. You can change the services it points to and the IP will remain, given you do not destroy this load balancer.
What I think I understand you would want is the capability to point a floating IP to a given load balancer (or another load balancer at another time). It is not possible at the moment, but I think it will be supported by Octavia load balancer at a later time. 
cc @jakubfovh

**Comment by kiorky** [comment]
Yes, i want to have like with many services at OVH the possibility to wire an IPFO which can be mapped on that LB or to another service (baremetal, other VM etc).

Idea behind the scene is that  as the IPFO remains over time,it's easier from a customer POV to map DNS configurations and other network rules on that ip fo, whatever service is really bound behind. This also make migrations easier as the IPFO is the only thing to switch from one service to another.

**Comment by errnesto** [comment]
> given you do not destroy this load balancer

I'm always really scared when I want to update or configure something on my ingress controller that it will be recreated and I'll loose my IP. Which means I have to inform external parties to update their configs.

But yes IPFO would be the perfect solution for this.

Or is there a way to keep the ovh loadbalancer even when the k8s resource is destroyed?

**Comment by JustDoItSascha** [comment]
Any updates on this?

**Comment by klbrvik** [comment]
Bump.

**Comment by LisaPerrier** [comment]
Hello @Tronix117, I am very sorry this happened. We don't not have a clear enough ETA right now but I'll make sure to update here as soon as I have one.

**Comment by JustDoItSascha** [comment]
@thomas-aw Better stick to GCP if you need this feature. Features in general coming in slow-mo here 🤣 If you don't need this feature, ovh is a decent cloud.

**Comment by thomas-aw** [comment]
I need it for health data hosting :rofl: 

I'm trying to use octavia ingress controller + nginx ingress controller but octavia doesn't support cert-manager ... 

@JustDoItSascha did you find any workaround ?

**Comment by JustDoItSascha** [comment]
@thomas-aw Yes, you have to create the load balancer yourself and assign the floating ip to it. You have to use "NodePort" as service typ in your nginx ingress controller. And then you have to create listeners for the load balancer to connect it to the ips of your nodes and there respective node ports of the service.

Works good, but you have to adjust the setup whenever you make changes to your nodes and for example add new machines.

And one pro tip from me: When you create the "members" for the load balancer, there is a timeout option, set to 5 seconds, set it to 10ms or something very low. They have network problems within the cluster and are not able to find the root cause. If you don't change it to a low timeout, your api or whatever will sometimes have long load times.

**Comment by thomas-aw** [comment]
@JustDoItSascha Awesome, many thanks. Do you have exemple of nodeport for nginx ingress controller ?

**Comment by JustDoItSascha** [comment]
I'm not sure what you mean? If you use an ingress controller in your kubernetes environment, for example nginx ingress controller, it will have an service for incoming requests. Per Default its a service of type "LoadBalancer" and it will automatically start a loadbalancer for you.

You have to change it to "NodePort", then it will assign a port for your service on every node of your cluster. The port is often something like 32xxxx. Then you start your load balancer manually and you reroute all the traffic to the port (e.g. 32222, whatever it will be on your cluster).

**Comment by JustDoItSascha** [comment]
Better: I could work for you! I'm a freelancer ;-)

**Comment by sashokbg** [comment]
Hello any updates from OVH team on this feature ?

**Comment by antonin-a** [comment]
Hello @sashokbg, it will be possible using OVHcloud Public Load Balancer (https://www.ovhcloud.com/fr/public-cloud/load-balancer/). The integration with MKS is currently in Alpha (you can join it be contacting us on our official Discord: https://discord.gg/ovhcloud) and will be in open Beta soon.

**Comment by antonin-a** [comment]
Hello, 
OVH Public Cloud Loadbalancers are now Generaly available for MKS clusters (default LB for MKS using k8s >= v1.31).
Those load balancers are using Floating IPs that can be attached/detached on Instances and LBs. 

You can also keep those floating IPs (and re use them later) in case of LB deletion using the following annotation
`loadbalancer.openstack.org/keep-floatingip= true`

https://help.ovhcloud.com/csm/en-public-cloud-kubernetes-expose-applications-using-load-balancer?id=kb_article_view&sysparm_article=KB0062878#supported-annotations-features

**Comment by jronda9** [comment]
Hello @antonin-a, is there a way do the same as "loadbalancer.openstack.org/load-balancer-address" but for a private IP address ? I want to create a private load balancer through Traefik in managed OVH cloud Kubernetes with a fixed private IP

**Comment by antonin-a** [comment]
Hello @jronda9 to use a specific private IP you should use `loadbalancer.openstack.org/port-id` describe on our documentation : https://help.ovhcloud.com/csm/en-public-cloud-kubernetes-expose-applications-using-load-balancer?id=kb_article_view&sysparm_article=KB0062878#supported-annotations-features

**Comment by jronda9** [comment]
Thank you @antonin-a  for your help, it works with the loadbalancer.openstack.org/port-id. 

I put some guidelines in case someone has same issue ;

1) Book the private ip you want to use for the LB in "Ports" from the network. It can be done in Horizon, by terraform (openstack_networking_port_v2) or command line (openstack port create).
2) Get the UUID of the private IP you booked
3) Create a load balancer service in kubernetes using annotation `loadbalancer.openstack.org/port-id: [UUID of private IP]`
